src/main/java/org/reaktivity/nukleus/kafka/internal/stream/KafkaClientProduceFactory.java

```diff
@@ -1551,13 +1578,16 @@ private void doEncodeRecordFin(
 
             encodeSlotBuffer.putBytes(encodeSlotLimit, encodeBuffer, 0, encodeProgress);
             encodeSlotLimit += encodeProgress;
-            encodeableRecordBytes += encodeProgress;
 
-            encodeableRecordCount++;
-
-            encodeableRecordBatchTimestampMax = Math.max(encodeableRecordBatchTimestamp, encodeableRecordTimestamp);
-
-            doSignalNextRequestIfNecessary(traceId);
+            if (encodeableRecordBytesDeferred > 0)
```

Comment:
Why would `encodeableRecordBytesDeferred` still be positive when `DATA` with `fin` arrives?
Shouldn't we be able to assert that this has become `0` after reducing it by the payload size as each `DATA` frame arrives?

Reply:
What this saying is following(probably it is not elegant and we can improve). If `encodeableRecordBytesDeferred>0` that means we already initiated doNetworkData which means that we don't need to signal since we want to complete a current record.

Reply:
Understood, but my main question was why we are explicitly assigning `encodeableRecordBytesDeferred = 0`.

That would seem to force `doSignalNextRequestIfNecessary(traceId)` on the next call to `doEncodeRecordCont` even if the record is not yet complete.

Reply:
I see what you mean yeah we should not set it 0.